File: frontend/test/favorites_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:bora_la/features/favorites/data/favorites_repository.dart';

void main() {
  test('fetchFavorites returns saved favorites', () async {
    final repository = FavoritesRepository();

    final favorites = await repository.fetchFavorites();

    expect(favorites, isNotEmpty);
    expect(favorites.first.name, 'Sábado de Música ao Vivo');
  });

  test('addFavorite creates a new favorite record', () async {
    final repository = FavoritesRepository();

    final favorite = await repository.addFavorite('evt_003', 'Noite de Jazz', 'Sede do Jazz');

    expect(favorite.eventId, 'evt_003');
    expect(favorite.name, 'Noite de Jazz');
  });
}

File: backend/test/bora_la/favorites_test.exs
defmodule BoraLa.FavoritesTest do
  use ExUnit.Case, async: true

  test "favorite creation payload builds expected structure" do
    payload = %{
      id: "fav_new",
      event_id: "evt_003",
      user_id: "user_001",
      name: "Noite de Jazz",
      location: "Sede do Jazz"
    }

    assert payload.event_id == "evt_003"
    assert payload.name == "Noite de Jazz"
  end
end
